ceph/ceph_admin/helper.py

```diff
@@ -481,6 +489,15 @@ def generate_snmp_spec(self, spec):
         engine_id = out.replace("-", "")
         if engine_id:
             spec["spec"]["engine_id"] = engine_id
+        # Check whether SNMPv2 or SNMPv3 credentials are present in spec
+        if "snmp_community" in spec["spec"]["credentials"]:
+            # Use SNMPv2
+            snmp_version = "V2c"
+            spec["spec"]["snmp_version"] = snmp_version
+        else:
```

Comment:
Add specific check like `snmp_v3_auth_username` and `snmp_v3_auth_password` not in `spec["spec"]["credentials"]`
 And in `else`  raise exception with unexepcted `credentials`